ql/src/test/queries/clientpositive/explainuser_2.q

```diff
@@ -6,6 +6,7 @@ set hive.strict.checks.bucketing=false;
 set hive.explain.user=true;
 set hive.metastore.aggregate.stats.cache.enabled=false;
 set hive.cbo.fallback.strategy=NEVER;
+set hive.optimize.merge.adjacent.union.distinct=false;
```

Comment:
Why do we disable it here? I am worried that the result of non-EXPLAIN could change

Reply:
hi @okumin, I've asked to keep the existing qtests unchanged as a regression when this feature is disabled + there is a new qtest added specifically to validate new feature. 
Note, you could check the outputs for those 2 qtests with enabled optimization in https://github.com/apache/hive/pull/5423/commits/74178b15f72bb8d53ea7b5fb934103c296f6c754#diff-f40b3c8ff967ea71b2e610dc10d9ed9a749f1054ad0d7144a3326f1b5820eb3f